### ISSUE TITLE
S3Mock fails when HTTPS is enabled on Java 11

### ISSUE THREAD

**Issue by cykl** [adobe/S3Mock#122]
Hi,

While upgrading a project to Java 11 all tests based on s3Mock started to fail with the following error:

```
    <error message="Unable to execute HTTP request: Remote host terminated the handshake" type="com.amazonaws.SdkClientException">com.amazonaws.SdkClientException: Unable to execute HTTP request: Remote host terminated the handshake
        at xxx.sdk.s3.S3ReportDescriptorTest.getInputStream_throws_FileNotFoundException_on_missing_key(S3ReportDescriptorTest.java:43)
Caused by: javax.net.ssl.SSLHandshakeException: Remote host terminated the handshake
        at xxx.sdk.s3.S3ReportDescriptorTest.getInputStream_throws_FileNotFoundException_on_missing_key(S3ReportDescriptorTest.java:43)
Caused by: java.io.EOFException: SSL peer shut down incorrectly
        at xxx.sdk.s3.S3ReportDescriptorTest.getInputStream_throws_FileNotFoundException_on_missing_key(S3ReportDescriptorTest.java:4
```

S3Client fails to communicate with the mock server. According to s3Mock logs, server is OK and listens for both HTTP and HTTPS request. After fiddling with JDKs and miscellaneous options it seems the issue is related to TLSv1.3 support which has been added to Java 11, see [release notes](https://www.oracle.com/technetwork/java/javase/11-relnote-issues-5012449.html#JDK-8145252).

 - I use `S3MockExtension` and invoking `withSecureConnection(false)` makes the tests pass. 
 - Enabling `-Djavax.net.debug=all -Djavax.net.ssl.trustStore=trustStore ` shown many error and warning related to TLSv1.3. 
 - Disabling TLSv1.3 with `-Djdk.tls.client.protocols=TLSv1.2` makes the tests pass.

Environment: 
```
$ java -version
openjdk version "11.0.2" 2019-01-15
OpenJDK Runtime Environment 18.9 (build 11.0.2+9)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.2+9, mixed mode)
```

**Comment by cykl** [comment]
Root cause could be:

```
javax.net.ssl|DEBUG|18|qtp28612052-24|2019-01-28 17:10:57.783 CET|CertificateMessage.java:998|Produced server Certificate message (
"Certificate": {
  "certificate_request_context": "",
  "certificate_list": [  
  {
    "certificate" : {
      "version"            : "v3",
      "serial number"      : "4D 89 4A 60",
      "signature algorithm": "SHA256withRSA",
      "issuer"             : "CN=localhost, OU=localhost, O=localhost, L=Unknown, ST=Unknown, C=Unknown",
      "not before"         : "2017-02-22 16:12:57.000 CET",
      "not  after"         : "2018-02-13 16:12:57.000 CET",
      "subject"            : "CN=localhost, OU=localhost, O=localhost, L=Unknown, ST=Unknown, C=Unknown",
      "subject public key" : "RSA",
      "extensions"         : [
        {
          ObjectId: 2.5.29.14 Criticality=false
          SubjectKeyIdentifier [
          KeyIdentifier [
          0000: F2 6E 10 96 EF 1E 3A EE   6C 1C 71 46 E8 E8 03 4D  .n....:.l.qF...M
          0010: C1 B3 9D F9                                        ....
          ]
          ]
        }
      ]}
    "extensions": {
      <no extension>
    }
  },
]
}
)
javax.net.ssl|ERROR|18|qtp28612052-24|2019-01-28 17:10:57.784 CET|TransportContext.java:313|Fatal (INTERNAL_ERROR): No preferred signature algorithm for CertificateVerify (
"throwable" : {
  javax.net.ssl.SSLException: No preferred signature algorithm for CertificateVerify
  	at java.base/sun.security.ssl.Alert.createSSLException(Alert.java:129)
  	at java.base/sun.security.ssl.Alert.createSSLException(Alert.java:117)
  	at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:308)
  	at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:264)
  	at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:255)
  	at java.base/sun.security.ssl.CertificateVerify$T13CertificateVerifyMessage.<init>(CertificateVerify.java:874)
  	at java.base/sun.security.ssl.CertificateVerify$T13CertificateVerifyProducer.onProduceCertificateVerify(CertificateVerify.java:1077)
  	at java.base/sun.security.ssl.CertificateVerify$T13CertificateVerifyProducer.produce(CertificateVerify.java:1070)
  	at java.base/sun.security.ssl.SSLHandshake.produce(SSLHandshake.java:436)
  	at java.base/sun.security.ssl.ClientHello$T13ClientHelloConsumer.goServerHello(ClientHello.java:1224)
  	at java.base/sun.security.ssl.ClientHello$T13ClientHelloConsumer.consume(ClientHello.java:1160)
  	at java.base/sun.security.ssl.ClientHello$ClientHelloConsumer.onClientHello(ClientHello.java:849)
  	at java.base/sun.security.ssl.ClientHello$ClientHelloConsumer.consume(ClientHello.java:810)
  	at java.base/sun.security.ssl.SSLHandshake.consume(SSLHandshake.java:392)
  	at java.base/sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:444)
  	at java.base/sun.security.ssl.SSLEngineImpl$DelegatedTask$DelegatedAction.run(SSLEngineImpl.java:1065)
  	at java.base/sun.security.ssl.SSLEngineImpl$DelegatedTask$DelegatedAction.run(SSLEngineImpl.java:1052)
  	at java.base/java.security.AccessController.doPrivileged(Native Method)
  	at java.base/sun.security.ssl.SSLEngineImpl$DelegatedTask.run(SSLEngineImpl.java:999)
  	at org.eclipse.jetty.io.ssl.SslConnection$DecryptedEndPoint.fill(SslConnection.java:511)
  	at org.eclipse.jetty.server.HttpConnection.fillRequestBuffer(HttpConnection.java:331)
  	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:243)
  	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:305)
  	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
  	at org.eclipse.jetty.io.ssl.SslConnection$DecryptedEndPoint.onFillable(SslConnection.java:411)
  	at org.eclipse.jetty.io.ssl.SslConnection.onFillable(SslConnection.java:305)
  	at org.eclipse.jetty.io.ssl.SslConnection$2.succeeded(SslConnection.java:159)
  	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
  	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:118)
  	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333)
  	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310)
  	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168)
  	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:126)
  	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:366)
  	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:765)
  	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:683)
  	at java.base/java.lang.Thread.run(Thread.java:834)}

)
```

**Comment by agudian** [comment]
Fixed with PR #123 by @cykl -- thanks a lot!

The release is on Docker hub and should show up on Maven central in a bit.
